src/current/v23.1/monitoring-and-alerting.md

```diff
@@ -189,6 +189,685 @@ If you rely on external tools for storing and visualizing your cluster's time-se
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+cockroach demo --geo-partitioned-replicas --insecure
+~~~
+
+By default, this geo-distributed demo cluster will not have any constraint violations.
+
+To introduce a violation that we can then query for, we'll modify the [zone configuration](configure-replication-zones.html) of the [`movr.users`](movr.html#the-movr-database) table.
+
+Let's use [`SHOW CREATE TABLE`](show-create.html) to see what existing zone configurations are attached to the `users` table, so we know what to modify.
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SHOW CREATE TABLE users;
+~~~
+
+~~~
+  table_name |                                    create_statement
+-------------+-----------------------------------------------------------------------------------------
+  users      | CREATE TABLE public.users (
+             |     id UUID NOT NULL,
+             |     city VARCHAR NOT NULL,
+             |     name VARCHAR NULL,
+             |     address VARCHAR NULL,
+             |     credit_card VARCHAR NULL,
+             |     CONSTRAINT users_pkey PRIMARY KEY (city ASC, id ASC)
+             | ) PARTITION BY LIST (city) (
+             |     PARTITION us_west VALUES IN (('seattle'), ('san francisco'), ('los angeles')),
+             |     PARTITION us_east VALUES IN (('new york'), ('boston'), ('washington dc')),
+             |     PARTITION europe_west VALUES IN (('amsterdam'), ('paris'), ('rome'))
+             | );
+             | ALTER PARTITION europe_west OF INDEX movr.public.users@users_pkey CONFIGURE ZONE USING
+             |     constraints = '[+region=europe-west1]';
+             | ALTER PARTITION us_east OF INDEX movr.public.users@users_pkey CONFIGURE ZONE USING
+             |     constraints = '[+region=us-east1]';
+             | ALTER PARTITION us_west OF INDEX movr.public.users@users_pkey CONFIGURE ZONE USING
+             |     constraints = '[+region=us-west1]'
+(1 row)
+~~~
+
+To create a constraint violation, let's use the [`ALTER PARTITION`](alter-partition.html) statement to tell the [ranges](architecture/overview.html#architecture-range) in the `europe_west` partition that they are explicitly supposed to *not* be in the `region=europe-west1` locality by issuing the following statement:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+ALTER PARTITION europe_west of INDEX movr.public.users@users_pkey CONFIGURE ZONE USING constraints = '[-region=europe-west1]';
+~~~
+
+Once the statement above executes, the ranges currently stored in that locality will now be in a state where they are explicitly not supposed to be in that locality, and are thus in violation of a constraint.
+
+In other words, we are telling the ranges "where you are now is exactly where you are *not* supposed to be". This will cause the cluster to rebalance the ranges, which will take some time. During the time it takes for the rebalancing to occur, the ranges will be in violation.
```

Comment:
```suggestion
In other words, this tells the ranges that "where you are now is exactly where you are **not** supposed to be". This will cause the cluster to rebalance the ranges, which will take some time. During the time it takes for the rebalancing to occur, the ranges will be in violation of a constraint.
```

Reply:
Fixed